src/frontend/src/flows/manage/deviceSettings.ts

```diff
@@ -0,0 +1,197 @@
+import { displayError } from "../../components/displayError";
+import { withLoader } from "../../components/loader";
+import { unreachable } from "../../utils/utils";
+import { footer } from "../../components/footer";
+import { DeviceData } from "../../../generated/internet_identity_types";
+import { hasOwnProperty } from "../../utils/utils";
+import { phraseRecoveryPage } from "../recovery/recoverWith/phrase";
+
+// The "device settings" page where users can view information about a device,
+// remove a device, make a recovery phrase protected, etc.
+
+// The styling of the page
+const style = () => html`<style></style> `;
+
+// Actual page content. We display the Identity Anchor and the list of
+// (non-recovery) devices. Additionally, if the user does _not_ have any
+// recovery devices, we display a warning "nag box" and suggest to the user
+// that they add a recovery device. If the user _does_ have at least one
+// recovery device, then we do not display a "nag box", but we list the
+// recovery devices.
+const pageContent = (
+  userNumber: bigint,
+  device: DeviceData,
+  isOnlyDevice: boolean
+) => html`
+  ${style()}
+  <div class="container">
+    <h1>Device Management</h1>
+    <p><strong>${device.alias}<strong></p>
+
+    ${
+      shouldOfferToProtect(device)
+        ? html`<button data-action="protect">Protect</button>
+            <p>You will be asked for your phrase</p>`
+        : ""
+    }
+
+    <button data-action="remove" ?disabled=${isOnlyDevice}>Delete Device</button>
+    ${
+      !isOnlyDevice && isProtected(device)
+        ? html`<p>
+            Your device is protected and you will be prompted to authenticate
+            with it before removal
+          </p>`
+        : ""
+    }
+    ${isOnlyDevice ? "You cannot remove your last device" : ""}
+    <button data-action="back">Back</button>
+  </div>
+  ${footer}
+`;
+
+// We offer to protect unprotected recovery phrases only, although in the
+// future we may offer to protect all devices
+const shouldOfferToProtect = (device: DeviceData): boolean =>
+  hasOwnProperty(device.purpose, "recovery") && !isProtected(device);
+
+const isProtected = (device: DeviceData): boolean =>
+  "protected" in device.protection;
+
+// Get the list of devices from canister and actually display the page
+export const deviceSettings = async (
+  userNumber: bigint,
+  connection: IIConnection,
+  device: DeviceData,
+  isOnlyDevice: boolean
+): Promise<void> => {
+  const container = document.getElementById("pageContent") as HTMLElement;
+
+  render(pageContent(userNumber, device, isOnlyDevice), container);
+  return init(userNumber, connection, device, isOnlyDevice);
+};
+
+// Get a connection that's authenticated with the given device
+// NOTE: this expects a recovery phrase device
+const deviceConnection = async (
+  userNumber: bigint,
+  device: DeviceData
+): Promise<IIConnection | null> => {
+  try {
+    const loginResult = await phraseRecoveryPage(userNumber, device);
```

Comment:
It would be nice, if you could pass on an intent here to make it more clear what's happening on the next page:
`Please provide your seed phrase in order to protect/delete it.`
or in case of recovery
`Please provide your seed phrase recover.`.

Reply:
I entered an invalid recovery phrase when protecting it and landed on the following screen:
![Screenshot 2022-06-30 at 14 42 13](https://user-images.githubusercontent.com/94825501/176679925-47ccf49f-2c9a-473c-a14f-b0d41dd39e68.png)

Reply:
Updated the error message:

> ... failed to authenticate using this seed phrase.